### PR TITLE
chore(Css/UI): Cleaned up UI elements to conform to mockups and standard component.

### PR DESCRIPTION
**Pull request by InfamousVague**
I will continue to work on this branch but I thought it'd be good to merge in what I have so far before continuing as it touches a lot of the codebase. 

Changes in this PR: 

Unify Create Group, Edit Group, Group Users, and Pinned Messages to use the same "Modal" component.
Edits Pinned messages to use native handlers for opening and closing as opposed to javscript, Does not touch the scrolling JS though I think we'll need to make it more complex in the future when looking for pinned messages not in history)

When experimental features are enabled this will currently render SOME of the communities code when you click "create community" which can be ignored for now it's just a start on the UI we'll need in the near future.

### CODE REVIEW

**Comment by InfamousVague** [comment]
image
image
image
image
image

**Comment by InfamousVague** [comment]
image

**Comment by phillsatellite** [comment]
I noticed small bug on MacOS  when you try to edit the Group username it will instead just close the edit group pop out so far PR is looking pretty good! Awesome feature to come! 

https://github.com/Satellite-im/Uplink/assets/93608357/5483ed7f-10df-46b6-89c9-b329d8a5d281

File: ui/src/main.rs [hunk 1]
```diff
@@ -1113,6 +1118,7 @@ fn AppNav<'a>(
             UplinkRoute::SettingsLayout {} => "/settings",
             UplinkRoute::FriendsLayout {} => "/friends",
             UplinkRoute::FilesLayout {} => "/files",
+            _ => "",
```
Comment (sdwoodbury) [inline ui/src/main.rs:1121]: should be "/community"

**Review by sdwoodbury — DISMISSED** [review]

**Comment by InfamousVague** [comment]
I did not want the community in the app navigation

File: ui/src/main.rs (hunk 1 above)
Comment (InfamousVague) [reply ui/src/main.rs:1121]: I did not want the community in the app navigation, I added this catch all for any routes that we don't care to add into the navigation.

**Comment by InfamousVague** [comment]
> I noticed small bug on MacOS when you try to edit the Group username it will instead just close the edit group pop out so far PR is looking pretty good! Awesome feature to come!
> Screen.Recording.2023-08-28.at.1.09.22.PM.mov

I'm not sure actually what the best way to handle this is, maybe just a z-index change i'll take a peek today.

**Comment by InfamousVague** [comment]
@phillsatellite the edit group name should be fixed now.

**Comment by phillsatellite** [comment]
Group name bug is fixed, thanks Matt! Once linter passes its approved by me 👍

**Review by sdwoodbury — APPROVED** [review]

**Comment by sdwoodbury** [comment]
rustfmt failed and needs the specified file to be formatted manually. looks like just deleting whitespace.

**Pull request merged**